compiler/rustc_mir/src/borrow_check/type_check/input_output.rs

```diff
@@ -80,6 +82,33 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
             let local = Local::new(argument_index + 1);
 
             let mir_input_ty = body.local_decls[local].ty;
+            // FIXME(jackh726): This is a hack. It's somewhat like
+            // `rustc_traits::normalize_after_erasing_regions`. Ideally, we'd
+            // like to normalize *before* inserting into `local_decls`, but I
+            // couldn't figure out where the heck that was.
```

Comment:
I found the following uses of `local_decls`. Not sure which if any are the ones you're looking for.

```
compiler/rustc_mir_build/src/build/mod.rs
792:    let mut local_decls = IndexVec::from_elem_n(LocalDecl::new(ty, span), 1);

compiler/rustc_mir/src/shim.rs
724:    let mut local_decls = local_decls_for_sig(&sig, span);
883:    let local_decls = local_decls_for_sig(&sig, span);

compiler/rustc_mir/src/transform/remove_zsts.rs
16:        let (basic_blocks, local_decls) = body.basic_blocks_and_local_decls_mut();

compiler/rustc_mir/src/transform/instcombine.rs
15:        let (basic_blocks, local_decls) = body.basic_blocks_and_local_decls_mut();

compiler/rustc_mir/src/transform/add_retag.rs
70:        let (basic_blocks, local_decls) = body.basic_blocks_and_local_decls_mut();

compiler/rustc_mir/src/transform/remove_unneeded_drops.rs
19:        let (basic_blocks, local_decls) = body.basic_blocks_and_local_decls_mut();

compiler/rustc_mir/src/transform/lower_slice_len.rs
27:    let (basic_blocks, local_decls) = body.basic_blocks_and_local_decls_mut();

compiler/rustc_mir/src/transform/deaggregator.rs
10:        let (basic_blocks, local_decls) = body.basic_blocks_and_local_decls_mut();
11:        let local_decls = &*local_decls;

compiler/rustc_mir/src/transform/lower_intrinsics.rs
15:        let (basic_blocks, local_decls) = body.basic_blocks_and_local_decls_mut();

compiler/rustc_mir/src/transform/match_branches.rs
51:        let (bbs, local_decls) = body.basic_blocks_and_local_decls_mut();

compiler/rustc_mir/src/transform/generator.rs
287:        let local_decls_len = body.local_decls.push(temp_decl);
387:    body.local_decls.raw[1].ty = ref_gen_ty;
402:    body.local_decls.raw[1].ty = pin_ref_gen_ty;
943:    body.local_decls[RETURN_PLACE] = LocalDecl::with_source_info(tcx.mk_unit(), source_info);
948:    body.local_decls[SELF_ARG] = LocalDecl::with_source_info(

compiler/rustc_mir/src/transform/promote_consts.rs
852:                promoted.local_decls[RETURN_PLACE] = LocalDecl::new(ty, span);
875:            let (blocks, local_decls) = self.source.basic_blocks_and_local_decls_mut();
```

Reply:
Yeah, that's the real question: which one of those is it? I looked through a couple, but didn't find it.

Reply:
Just for reference, without this, `normalization-generality` fails with:
```
error: internal compiler error: broken MIR in DefId(0:19 ~ normalization_generality[317d]::slice::{closure#0}) (bb0[0]): equate_normalized_input_or_output: `&'_#2r [slice::{closure#0} closure_kind_ty=i8 closure_sig_as_fn_ptr_ty=for<'r> extern "rust-call" fn((&ReLateBound(DebruijnIndex(0), BoundRegion { var: 0, kind: BrNamed('r) }) (), &ReLateBound(DebruijnIndex(0), BoundRegion { var: 0, kind: BrNamed('r) }) ())) -> &ReLateBound(DebruijnIndex(0), BoundRegion { var: 0, kind: BrNamed('r) }) () upvar_tys=()]==&'_#9r [slice::{closure#0} closure_kind_ty=i8 closure_sig_as_fn_ptr_ty=for<'r> extern "rust-call" fn((&ReLateBound(DebruijnIndex(0), BoundRegion { var: 0, kind: BrNamed('r) }) (), &ReLateBound(DebruijnIndex(0), BoundRegion { var: 0, kind: BrNamed('r) }) ())) -> <&'_#10r () as Yokeable<ReLateBound(DebruijnIndex(0), BoundRegion { var: 0, kind: BrNamed('r) })>>::Output upvar_tys=()]` failed with `NoSolution`
```